### ISSUE TITLE
Unusual behavior with substr/decode_prefix in while scope?

### ISSUE THREAD

**Issue by PaulGWebster** [rurban/Cpanel-JSON-XS#82]
## Console:
$ perl bug.pl
Buffer: [0][1][2][3]
Offset/size: 3
New buffer: [1][2][3]

Buffer: [1][2][3]
substr outside of string at bug.pl line 11.
Offset/size: 2275
Use of uninitialized value $buffer in concatenation (.) or string at bug.pl line 13.
New buffer:

## Code:

```
use warnings;
use strict;

use Cpanel::JSON::XS;

my $buffer = "[0][1][2][3]";

while ($buffer) {
        print "Buffer: $buffer\n";
        my ($data,$size) = Cpanel::JSON::XS->new->decode_prefix($buffer);
        $buffer = substr($buffer,$size);
        print "Offset/size: $size\n";
        print "New buffer: $buffer\n\n";
}
```

**Comment by rurban** [comment]
Thanks for the testcase. The internal sv string buffer may get re-allocated, and then the offset calculation (=> size) is totally wrong.

**Comment by rurban** [comment]
Fixed with 3.0228
